### ISSUE TITLE
Live Streams

### ISSUE THREAD

**Issue by CrazyWolf13** [fe80Grau/ytdlp2STRM#6]
Theoretically yt-dlp works with many different video hosting platforms, even with twitch.tv or youtube live streams, does ytdlp2strm work with live streams? Has this ever been tested?

**Comment by Traace** [comment]
I like to know that, too.

I want to convert 24/7 Live Streams into STRM for jellyfin 👯

**Comment by CrazyWolf13** [comment]
@fe80Grau  Any idea?

**Comment by CrazyWolf13** [comment]
Wow, this is amazing!
Can't wait to test it:)

**Comment by fe80Grau** [comment]
You can test Twitch live stream with the last commit. 
I added more instruction in readme but i put here a litte instructions.

restart service after git pull 
Copy channel_list.json (older version) yo ./plugins/youtube/channel_list.json
Is not necessary edit the older cronjob (i added compatibility with last verions), but i recommend it to keep an order
mkdir  for strm Twitch (example; mkdir /media/Twitch)
Edit config and channel_list in plugins/twitch
You can test it with
`cd /opt/ytdlp2STRM && /usr/bin/python3 /opt/ytdlp2STRM/cli.py --m plugins.twitch.to_strm --p twitch,direct`
If you want to run this every 10 minutes:
> ``` console
> cd /etc/cron.d && sudo echo "*/10 * * * * root cd /opt/ytdlp2STRM && /usr/bin/python3 /opt/ytdlp2STRM/cli.py --m plugins.twitch.to_strm --p twitch,direct" > ytdlp2STRM_twitch_direct
> ```

**Comment by CrazyWolf13** [comment]
Wow, amazing!
Been a little bit busy in last time, but can't wait to test it:)

**Comment by Traace** [comment]
Streams are not working. I think Youtube is blocking my Server WAN IP.

```
06/06/2023 09:33:25
Running plugins.youtube.to_strm with ['youtube', 'direct'] params
Preparing channel @uproxx/streams
No channel ID Found, Research with no video tab
yt-dlp --compat-options no-youtube-channel-redirect --compat-options no-youtube-unavailable-videos --restrict-filenames --ignore-errors --no-warnings --playlist-start 1 --playlist-end 1 --print channel_url https://www.youtube.com/@uproxx/streams
Cleaning /media/Youtube/@uproxx [False] folder...
Inflating nfo file..
No poster detected
No landscape detected
Command 
 yt-dlp https://www.youtube.com/channel/False --write-description --playlist-items 0 --output "/media/Youtube/ERROR: [youtube:tab] False: Unable to download API page: HTTP Error 400: Bad Request (caused by <HTTPError 400: 'Bad Request'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U.description" > /dev/null 2>&1 && cat "/media/Youtube/ERROR: [youtube:tab] False: Unable to download API page: HTTP Error 400: Bad Request (caused by <HTTPError 400: 'Bad Request'>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U.description"
Output 
 
Processing videos in channel
{'video_id': 'ERROR: [youtube] 05689ErDUdM: No video formats found!', 'video_name': ' please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U [ERROR: [youtube] 05689ErDUdM: No video formats found!]'}
```

Any way to connect via proxy?

**Comment by fe80Grau** [comment]
hi @Traace , I'm getting properly the streams tab:
```
Preparing channel @uproxx/streams
Cleaning D:/Youtube/@uproxx [UCI2HI_aQ_S3v4vfDQgNtK1g] folder...
Inflating  nfo file..
Descriptions only works in Linux system at the moment
Processing videos in channel
{'video_id': '05689ErDUdM', 'video_name': 'Rap Live Radio 24/7 | Hip-Hop, Popular Rap Music & UPROXX Sessions Live Performances! 2023-06-06 13:22
5689ErDUdM]'}
```

Can you check your yt-dlp version with yt-dlp -U command?
You need last stable version 
```
Available version: stable@2023.03.04, Current version: stable@2023.03.04
yt-dlp is up to date (stable@2023.03.04)
```

**Comment by Traace** [comment]
Yes, sorry maybe there was a misunderstanding. 

I know that yt-dlp returns 'No video formats found' when the IP was blocked by youtube. 
This is no issue with your code at all :) (btw: many thanks for the work)

Proxy support would be awesome, indeed. Meanwhile as a workaround I can send that traffic through a VPN tunnel.

**Comment by fe80Grau** [comment]
Yep @Traace , I added proxy feature for youtube plugin. You can check it in last commit. Add to your plugins/youtube/config.json proxy and proxy_url params like

```
{
    "strm_output_folder" : "D:/Youtube",
    "channels_list_file" : "./plugins/youtube/channel_list.json",
    "days_dateafter" : "0", 
    "videos_limit" : "10",
    "sponsorblock" : false,
    "sponsorblock_cats" : "sponsor",
    "proxy" : true,
    "proxy_url" : "socks5://user:pass@127.0.0.1:1080/"
}
```

Let me know if you try this.

**Comment by Traace** [comment]
Hey @fe80Grau, thanks again. The Proxy works :)

I've noticed that since i added the proxy param, while creating the .strm file the following gets logged:

```
Inflating  nfo file..
Output 
 cat: unrecognized option '--proxy'
Try 'cat --help' for more information.
```

That warning also gets written into the tvshow.nfo plot section and is thus visible in jellyfin:
```
 <plot>cat: unrecognized option '--proxy'
Try 'cat --help' for more information.</plot>
```

**Comment by fe80Grau** [comment]
I think that is fixed with last commit.

**Comment by Traace** [comment]
Yes, nfo and strm files are fine now. But the stream can't be played in jellyfin because the IP is blocked. Google is known for blocking many server networks around the world.

I could route jellyfin and ffmpeg outgoing traffic via proxy, too. But that would also route more than 3000 LiveTV Channels through the proxy... I dont want that to happen :)

Routing just the ipv4 from manifest.googlevideo.com through a VPN does fix nothing. I guess they are using a CDN Backend with many different IP's.

Can't find another solution... Maybe my infrastucture is to big for this :)

**Comment by fe80Grau** [comment]
Okaaay, direct mode is not good for you because ytdlp2strm will do a redirect to video url and Jellyfin process this stream trying to connect to Youtube servers, i think that you can wok with bridge mode with streams,  the proxy command is setted to yt-dlp command to extract bytes from stream (jumping google restrictions), this bytes will be storage in local file, this local file is served by Jellyfin and never request anything from youtube 

Pull last comit i fixed a little param in yt-dlp for bridge mode.

Is simple to test, edit your strm file and change /direct/ to /bridge/ , http://127.0.0.1:5000/youtube/direct/HPiv1pzIPRc -> http://127.0.0.1:5000/youtube/bridge/HPiv1pzIPRc
Can you try and tell me?

**Comment by fe80Grau** [comment]
@Traace i close this because the Live stream feature is implemmented now. If you need more help with your case you can open a new issue

Thank you for all tests 😄
